Index: collective.solr/trunk/collective/solr/solr.py
===================================================================
--- collective.solr/trunk/collective/solr/solr.py	(revision 13421)
+++ collective.solr/trunk/collective/solr/solr.py	(revision 13422)
@@ -15,21 +15,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-# $Id$
 # A simple Solr client for python.
-# This is prototype level code and subject to change.
 #
 # quick examples on use:
 #
 # from collective.solr.solr import *
 # c = SolrConnection(host='localhost:8983', persistent=True)
 # c.add(id='500',name='python test doc')
+# c.commit()
+# c.search(q='id:[* TO *]', wt='python', rows='10',indent='on')
 # c.delete('123')
 # c.commit()
-# print c.search(q='id:[* TO *]', wt='python', rows='10',indent='on')
-# data = c.search(q='id:500', wt='python')
-# print 'first match=', eval(data)['response']['docs'][0]
 
 import httplib
 import socket
