Index: collective.solr/trunk/collective/solr/tests/test_parser.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_parser.py	(revision 13467)
+++ collective.solr/trunk/collective/solr/tests/test_parser.py	(revision 13468)
@@ -83,7 +83,7 @@
 
     def testParseConfig(self):
         schema = SolrSchema(getData('schema.xml'))
-        fields = schema.fields
+        fields = schema
         self.assertEqual(len(fields), 17) # 17 fields defined in schema.xml
         self.assertEqual(fields['id'].type, 'string')
         self.assertEqual(fields['id'].class_, 'solr.StrField')
Index: collective.solr/trunk/collective/solr/parser.py
===================================================================
--- collective.solr/trunk/collective/solr/parser.py	(revision 13467)
+++ collective.solr/trunk/collective/solr/parser.py	(revision 13468)
@@ -87,7 +87,7 @@
             raise AttributeError, name
 
 
-class SolrSchema(object):
+class SolrSchema(SolrField):
     """ a solr schema parser:  the xml schema is partially parsed and the
         information collected is later on used both for indexing items as
         well as buiding search queries;  for the time being we are only
@@ -108,7 +108,6 @@
         if isinstance(data, basestring):
             data = StringIO(data)
         types = {}
-        fields = self.fields = {}
         for action, elem in iterparse(data):
             name = elem.get('name')
             if elem.tag == 'fieldType':
@@ -117,5 +116,5 @@
                 field = SolrField(types[elem.get('type')])
                 field.update(elem.attrib)
                 field['class_'] = field['class']    # `.class` will not work
-                fields[name] = field
+                self[name] = field
 
