Index: collective.solr/trunk/collective/solr/parser.py
===================================================================
--- collective.solr/trunk/collective/solr/parser.py	(revision 13460)
+++ collective.solr/trunk/collective/solr/parser.py	(revision 13461)
@@ -74,12 +74,17 @@
         return self
 
 
-# Parse config
+class SolrSchema(object):
+    """ a solr schema parser:  the xml schema is partially parsed and the
+        information collected is later on used both for indexing items as
+        well as buiding search queries;  for the time being we are only
+        interested in explicitly defined fields and their data types, so
+        all <analyzer> (tokenizers, filters) and <dynamicField> information
+        is ignored;  all the other fields like <uniqueKey>, <copyField>,
+        <solrQueryParser>, <defaultSearchField> etc get ignored as well,
+        since they are only used by solr, but not relevant when building
+        search or indexing queries """
 
-# handle dynamic fields?
-# uniqueKey
-# solrQueryParser
-#  copyField - should remove dest from fields and add source if not already present - set source to same type as dest
 
 def parseConfig(data):
     """Parse schema.xml"""
