Index: collective.solr/trunk/collective/solr/tests/test_indexablewrapper.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_indexablewrapper.py	(revision 13459)
+++ collective.solr/trunk/collective/solr/tests/test_indexablewrapper.py	(revision 13460)
@@ -15,7 +15,6 @@
 from collective.solr.indexablewrapper import SolrExtensibleIndexableObjectWrapper
 
 
-
 @onsetup
 def setup_product():
     """Set up additional products and ZCML required to test this product.
@@ -42,7 +41,6 @@
 ptc.setupPloneSite()
 
 
-
 class TestCase(ptc.PloneTestCase):
     pass
 
