Index: collective.solr/trunk/collective/solr/tests/test_solr.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_solr.py	(revision 13492)
+++ collective.solr/trunk/collective/solr/tests/test_solr.py	(revision 13493)
@@ -67,6 +67,54 @@
         self.failUnlessEqual(node.text, '55')
         res.find('QTime')
 
+    def test_optimize(self):
+        commit_request = getData('optimize_request.txt')
+        commit_response = getData('commit_response.txt')
+        output = []
+
+        c = SolrConnection(host='localhost:8983', persistent=True)
+        fakehttp(c, commit_response, output)
+        res = c.commit(optimize=True)
+
+        output = ''.join(output).replace('\r','')
+        self.failUnlessEqual(output, commit_request)
+
+    def test_commit_no_wait_flush(self):
+        commit_request = getData('commit_request.txt')
+        commit_response = getData('commit_response.txt')
+        output = []
+
+        c = SolrConnection(host='localhost:8983', persistent=True)
+        fakehttp(c, commit_response, output)
+        res = c.commit(waitFlush=False)
+
+        output = ''.join(output).replace('\r','')
+        self.failUnlessEqual(output, commit_request)
+
+    def test_commit_no_wait_searcher(self):
+        commit_request = getData('commit_request_no_wait_searcher.txt')
+        commit_response = getData('commit_response.txt')
+        output = []
+
+        c = SolrConnection(host='localhost:8983', persistent=True)
+        fakehttp(c, commit_response, output)
+        res = c.commit(waitSearcher=False)
+
+        output = ''.join(output).replace('\r','')
+        self.failUnlessEqual(output, commit_request)
+
+    def test_commit_no_wait(self):
+        commit_request = getData('commit_request_no_wait.txt')
+        commit_response = getData('commit_response.txt')
+        output = []
+
+        c = SolrConnection(host='localhost:8983', persistent=True)
+        fakehttp(c, commit_response, output)
+        res = c.commit(waitFlush=False, waitSearcher=False)
+
+        output = ''.join(output).replace('\r','')
+        self.failUnlessEqual(output, commit_request)
+
     def test_search(self):
         search_request = getData('search_request.txt')
         search_response = getData('search_response.txt')
Index: collective.solr/trunk/collective/solr/tests/data/optimize_request.txt
===================================================================
--- collective.solr/trunk/collective/solr/tests/data/optimize_request.txt	(revision 0)
+++ collective.solr/trunk/collective/solr/tests/data/optimize_request.txt	(revision 13493)
@@ -0,0 +1,7 @@
+POST /solr/update HTTP/1.1
+Host: localhost
+Accept-Encoding: identity
+Content-Length: 11
+Content-Type: text/xml; charset=utf-8
+
+<optimize/>
\ No newline at end of file

Property changes on: collective.solr/trunk/collective/solr/tests/data/optimize_request.txt
___________________________________________________________________
Name: svn:eol-style
   + native

Index: collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait_searcher.txt
===================================================================
--- collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait_searcher.txt	(revision 0)
+++ collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait_searcher.txt	(revision 13493)
@@ -0,0 +1,7 @@
+POST /solr/update HTTP/1.1
+Host: localhost
+Accept-Encoding: identity
+Content-Length: 30
+Content-Type: text/xml; charset=utf-8
+
+<commit waitSearcher="false"/>
\ No newline at end of file

Property changes on: collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait_searcher.txt
___________________________________________________________________
Name: svn:eol-style
   + native

Index: collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait.txt
===================================================================
--- collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait.txt	(revision 0)
+++ collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait.txt	(revision 13493)
@@ -0,0 +1,7 @@
+POST /solr/update HTTP/1.1
+Host: localhost
+Accept-Encoding: identity
+Content-Length: 48
+Content-Type: text/xml; charset=utf-8
+
+<commit waitFlush="false" waitSearcher="false"/>
\ No newline at end of file

Property changes on: collective.solr/trunk/collective/solr/tests/data/commit_request_no_wait.txt
___________________________________________________________________
Name: svn:eol-style
   + native

