Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 0)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13472)
@@ -0,0 +1,36 @@
+from zope.component import adapts, getMultiAdapter
+from Products.CMFCore.utils import getToolByName
+
+from DateTime import DateTime
+
+from collective.solr.solr import SolrConnection
+
+from collective.solr.indexablewrapper import ISolrIndexableObjectWrapper
+
+def datehandler(value):
+    # We might want to handle datetime and time as well
+    # XXX check the enfold.solr implementation
+    if isinstance(value, DateTime):
+        return value.toZone('UTC')
+    return 'null'
+
+handlers = {'solr.DateField': datehandler}
+
+def indexObject(obj):
+    portal = getToolByName(obj, 'portal_url').getPortalObject()
+    wrapper = getMultiAdapter((obj,portal), ISolrIndexableObjectWrapper)
+    conn = SolrConnection(host='localhost:8983', persistent=True)
+    data = {}
+    schema = conn.getSchema()
+    for key, class_ in schema.fields.items():
+        value = getattr(wrapper, key, None)
+        if callable(value):
+            value = value()
+        if handlers.has_key(class_):
+            value = handlers[class_](value)
+        data[key] = value
+    conn.add(**data)
+    r = conn.commit()
+
+# from collective.solr.indexer import indexObject
+# indexObject(app.eli.nordic)
\ No newline at end of file

Property changes on: collective.solr/trunk/collective/solr/indexer.py
___________________________________________________________________
Name: svn:eol-style
   + native

