Index: collective.solr/trunk/collective/solr/tests/test_solr.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_solr.py	(revision 13483)
+++ collective.solr/trunk/collective/solr/tests/test_solr.py	(revision 13484)
@@ -6,22 +6,23 @@
 from collective.solr.tests.utils import getData
 
 
-class TestSolr(TestCase):
+def fakehttp(solrconn, fakedata, output):
+    class FakeSocket(StringIO):
+        def sendall(self, str): output.append(str)
+        def makefile(self, mode, name): return self
+        def read(self, amt=None):
+            if self.closed: return ''
+            return StringIO.read(self, amt)
+        def readline(self, length=None):
+            if self.closed: return ''
+            return StringIO.readline(self, length)
+    class FakeHTTPConnection(HTTPConnection):
+        def connect(self):
+            self.sock = FakeSocket(fakedata)
+    solrconn.conn = FakeHTTPConnection(solrconn.conn.host)
+
 
-    def fakehttp(self, solrconn, fakedata, output):
-        class FakeSocket(StringIO):
-            def sendall(self, str): output.append(str)
-            def makefile(self, mode, name): return self
-            def read(self, amt=None):
-                if self.closed: return ''
-                return StringIO.read(self, amt)
-            def readline(self, length=None):
-                if self.closed: return ''
-                return StringIO.readline(self, length)
-        class FakeHTTPConnection(HTTPConnection):
-            def connect(self):
-                self.sock = FakeSocket(fakedata)
-        solrconn.conn = FakeHTTPConnection(solrconn.conn.host)
+class TestSolr(TestCase):
 
     def test_add(self):
         add_request = getData('add_request.txt')
@@ -29,7 +30,7 @@
         output = []
 
         c = SolrConnection(host='localhost:8983', persistent=True)
-        self.fakehttp(c, add_response, output)
+        fakehttp(c, add_response, output)
         res = c.add(id='500',name='python test doc')
 
         output = ''.join(output).replace('\r','')
@@ -50,7 +51,7 @@
         output = []
 
         c = SolrConnection(host='localhost:8983', persistent=True)
-        self.fakehttp(c, commit_response, output)
+        fakehttp(c, commit_response, output)
         res = c.commit()
 
         output = ''.join(output).replace('\r','')
@@ -72,7 +73,7 @@
         output = []
 
         c = SolrConnection(host='localhost:8983', persistent=True)
-        self.fakehttp(c, search_response, output)
+        fakehttp(c, search_response, output)
         res = c.search(q='id:[* TO *]', wt='xml', rows='10',indent='on')
 
         output = ''.join(output).replace('\r','')
@@ -86,7 +87,7 @@
         output = []
 
         c = SolrConnection(host='localhost:8983', persistent=True)
-        self.fakehttp(c, delete_response, output)
+        fakehttp(c, delete_response, output)
         res = c.delete('500')
 
         output = ''.join(output).replace('\r','')
