Index: collective.solr/trunk/collective/solr/solr.py
===================================================================
--- collective.solr/trunk/collective/solr/solr.py	(revision 13464)
+++ collective.solr/trunk/collective/solr/solr.py	(revision 13465)
@@ -35,7 +35,7 @@
 from xml.sax.saxutils import escape
 import codecs
 import urllib
-
+from collective.solr.parser import SolrSchema
 
 class SolrException(Exception):
     """ An exception thrown by solr connections """
@@ -198,16 +198,7 @@
         return parsed
 
     def getSchema(self):
-        def _parseElement(el):
-            el_dict = {}
-            fields = ('indexed', 'stored', 'type', 'required')
-            for field in fields:
-                el_dict[field] = el.get(field)
-            return el.get('name'), el_dict
-
-        url = self.solrBase + '/admin/get-file.jsp?file=schema.xml'
+        url = '%s/admin/get-file.jsp?file=schema.xml' % self.solrBase
         self.conn.request('GET', url)
         xml = self.__errcheck(self.conn.getresponse()).read()
-        xml = StringIO(xml.strip())
-        elem = xml_parse(xml).getroot()
-        return [_parseElement(el) for el in elem.findall('fields/field')]
+        return SolrSchema(xml.strip())
