Index: collective.solr/trunk/collective/solr/parser.py
===================================================================
--- collective.solr/trunk/collective/solr/parser.py	(revision 13432)
+++ collective.solr/trunk/collective/solr/parser.py	(revision 13433)
@@ -148,7 +148,6 @@
     stack.append(result)
 
 def handle_start_arr(elem, stack):
-    # Set special attributes like numfound and start
     arr = []
     add_node(arr, elem, stack)
     stack.append(arr)
