Index: collective.solr/trunk/setup.py
===================================================================
--- collective.solr/trunk/setup.py	(revision 13416)
+++ collective.solr/trunk/setup.py	(revision 13417)
@@ -25,7 +25,6 @@
       zip_safe=False,
       install_requires=[
           'setuptools',
-          'simplejson',
           # -*- Extra requirements: -*-
       ],
       entry_points="""

Property changes on: collective.solr/trunk
___________________________________________________________________
Name: svn:ignore
   + build
dist
*.egg-info


