Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 13475)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13476)
@@ -1,12 +1,10 @@
-from zope.component import adapts, getMultiAdapter
+from zope.component import getMultiAdapter
 from Products.CMFCore.utils import getToolByName
-
 from DateTime import DateTime
-
 from collective.solr.solr import SolrConnection
-
 from collective.solr.indexablewrapper import ISolrIndexableObjectWrapper
 
+
 def datehandler(value):
     # We might want to handle datetime and time as well
     # XXX check the enfold.solr implementation
@@ -14,8 +12,10 @@
         return value.toZone('UTC')
     return 'null'
 
+
 handlers = {'solr.DateField': datehandler}
 
+
 def indexObject(obj):
     portal = getToolByName(obj, 'portal_url').getPortalObject()
     wrapper = getMultiAdapter((obj,portal), ISolrIndexableObjectWrapper)
@@ -33,4 +33,5 @@
     r = conn.commit()
 
 # from collective.solr.indexer import indexObject
-# indexObject(app.eli.nordic)
\ No newline at end of file
+# indexObject(app.eli.nordic)
+
Index: collective.solr/trunk/collective/solr/solr.py
===================================================================
--- collective.solr/trunk/collective/solr/solr.py	(revision 13475)
+++ collective.solr/trunk/collective/solr/solr.py	(revision 13476)
@@ -29,16 +29,16 @@
 
 import httplib
 import socket
-from StringIO import StringIO 
 from elementtree.ElementTree import fromstring
-from elementtree.ElementTree import parse as xml_parse
 from xml.sax.saxutils import escape
 import codecs
 import urllib
 from collective.solr.parser import SolrSchema
 
+
 class SolrException(Exception):
     """ An exception thrown by solr connections """
+
     def __init__(self, httpcode, reason=None, body=None):
         self.httpcode = httpcode
         self.reason = reason
@@ -53,6 +53,7 @@
 
 
 class SolrConnection:
+
     def __init__(self, host='localhost:8983', solrBase='/solr', persistent=True, postHeaders={}):
         self.host = host
         self.solrBase = solrBase
@@ -202,3 +203,4 @@
         self.conn.request('GET', url)
         xml = self.__errcheck(self.conn.getresponse()).read()
         return SolrSchema(xml.strip())
+
