Index: collective.solr/trunk/collective/solr/tests/test_integration.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_integration.py	(revision 13526)
+++ collective.solr/trunk/collective/solr/tests/test_integration.py	(revision 13527)
@@ -5,7 +5,6 @@
 from unittest import TestSuite, makeSuite, main
 from Products.Five import zcml
 from Products.Five import fiveconfigure
-from Testing import ZopeTestCase as ztc
 from Products.PloneTestCase import PloneTestCase as ptc
 from Products.PloneTestCase.layer import onsetup
 
@@ -21,7 +20,7 @@
 ptc.setupPloneSite(products=['collective.solr'])
 
 
-# test-specific import go here...
+# test-specific imports go here...
 from zope.component import queryUtility, getUtilitiesFor
 from collective.indexing.interfaces import IIndexQueueProcessor
 from collective.solr.interfaces import ISolrIndexQueueProcessor
@@ -42,6 +41,12 @@
         self.failUnless(IIndexQueueProcessor.providedBy(proc))
         self.failUnless(ISolrIndexQueueProcessor.providedBy(proc))
 
+    def testRegisteredProcessors(self):
+        procs = list(getUtilitiesFor(IIndexQueueProcessor))
+        self.failUnless(procs, 'no utilities found')
+        solr = queryUtility(ISolrIndexQueueProcessor)
+        self.failUnless(solr in [util for name, util in procs], 'solr utility not found')
+
 
 def test_suite():
     return TestSuite([
