Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 13519)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13520)
@@ -39,7 +39,7 @@
         self.index(obj, attributes)
 
     def unindex(self, obj):
-        data = self.getData(obj)
+        data = self.getData(obj, attributes=['id'])
         self.prepareData(data)
         # TODO: perhaps we should consider <uniqueKey> here
         assert data.has_key('id'), "no id in object data"
@@ -53,16 +53,20 @@
 
     # helper methods
 
-    def getData(self, obj):
+    def getData(self, obj, attributes=None):
+        schema = self.getSchema()
+        if attributes is None:
+            attributes = schema.keys()
         data, marker = {}, []
-        for name, field in self.getSchema().items():
+        for name in attributes:
             value = getattr(obj, name, marker)
             if value is marker:
                 continue
             if callable(value):
                 value = value()
-            if handlers.has_key(field.class_):
-                value = handlers[field.class_](value)
+            handler = handlers.get(schema[name].class_, None)
+            if handler is not None:
+                value = handler(value)
             data[name] = value
         return data
 
