Index: collective.solr/trunk/collective/solr/tests/test_integration.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_integration.py	(revision 13530)
+++ collective.solr/trunk/collective/solr/tests/test_integration.py	(revision 13531)
@@ -65,7 +65,7 @@
         fakehttp(connection, response, output)      # fake add response
         self.folder.processForm({'title': 'Foo'})   # updating sends data
         output = ''.join(output).replace('\r', '')
-        # TODO: collective.solr still need to be changed to put the actual
+        # TODO: collective.solr still needs to be changed to put the actual
         # objects into the queue instead of the UIDs (which will make things
         # clearer anyway), so for now we check for the UID... and also only
         # for it's lower-case version, since this is mangled somewhere on
