Index: collective.solr/trunk/collective/solr/tests/test_indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_indexer.py	(revision 13516)
+++ collective.solr/trunk/collective/solr/tests/test_indexer.py	(revision 13517)
@@ -45,6 +45,15 @@
         output = ''.join(output).replace('\r', '')
         self.assertEqual(output, getData('add_request.txt'))
 
+    def testUnindexObject(self):
+        proc = self.setupProcessor()
+        output = []
+        response = getData('delete_response.txt')
+        fakehttp(proc.getConnection(), response, output)    # fake response
+        proc.unindex(Foo(id='500', name='python test doc')) # unindexing sends data
+        output = ''.join(output).replace('\r', '')
+        self.assertEqual(output, getData('delete_request.txt'))
+
 
 def test_suite():
     return TestSuite([
Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 13516)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13517)
@@ -39,7 +39,11 @@
         self.index(obj, attributes)
 
     def unindex(self, obj):
-        pass
+        data = self.getData(obj)
+        self.prepareData(data)
+        # TODO: perhaps we should consider <uniqueKey> here
+        assert data.has_key('id'), "no id in object data"
+        self.getConnection().delete(id=data['id'])
 
     def begin(self):
         pass
