Index: collective.solr/trunk/collective/solr/tests/test_indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/tests/test_indexer.py	(revision 13513)
+++ collective.solr/trunk/collective/solr/tests/test_indexer.py	(revision 13514)
@@ -19,17 +19,20 @@
         SolrIndexQueueProcessor().prepareData(data)
         self.assertEqual(data, {'allowedRolesAndUsers': ['user$test_user_1_', 'user$portal_owner']})
 
-    def testIndexObject(self):
-        foo = Foo(id='500', name='python test doc')
+    def setupProcessor(self):
         proc = SolrIndexQueueProcessor()
         proc.setHost()
         conn = proc.getConnection()
         fakehttp(conn, getData('schema.xml'), [])   # fake schema response
         proc.getSchema()                            # read and cache the schema
+        return proc
+
+    def testIndexObject(self):
+        proc = self.setupProcessor()
         output = []
         response = getData('add_response.txt')
-        fakehttp(conn, response, output)            # fake add response
-        proc.index(foo)                             # indexing sends data
+        fakehttp(proc.getConnection(), response, output)    # fake add response
+        proc.index(Foo(id='500', name='python test doc'))   # indexing sends data
         output = ''.join(output).replace('\r', '')
         self.assertEqual(output, getData('add_request.txt'))
 
