Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 13555)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13556)
@@ -35,7 +35,7 @@
         conn = self.getConnection()
         if conn is not None:
             data = self.getData(obj, attributes)
-            self.prepareData(data)
+            self.prepareData(data, obj)
             try:
                 conn.add(**data)
             except SolrException, e:
@@ -48,7 +48,7 @@
         conn = self.getConnection()
         if conn is not None:
             data = self.getData(obj, attributes=['id'])
-            self.prepareData(data)
+            self.prepareData(data, obj)
             # TODO: perhaps we should consider <uniqueKey> here
             assert data.has_key('id'), "no id in object data"
             try:
@@ -86,9 +86,12 @@
             data[name] = value
         return data
 
-    def prepareData(self, data):
+    def prepareData(self, data, obj=None):
         """ modify data according to solr specifics, i.e. replace ':' by '$'
             for "allowedRolesAndUsers" etc """
+        uid = data.get('uid', None)
+        if uid is None and hasattr(obj, 'UID'):
+            data['uid'] = obj.UID()     # TODO: fix this asap!!
         allowed = data.get('allowedRolesAndUsers', None)
         if allowed is not None:
             data['allowedRolesAndUsers'] = [r.replace(':','$') for r in allowed]
