Index: collective.solr/trunk/TODO.txt
===================================================================
--- collective.solr/trunk/TODO.txt	(revision 13616)
+++ collective.solr/trunk/TODO.txt	(revision 13617)
@@ -1,7 +1,9 @@
 TODOs:
 * browsertest for indexing
+* browser view for reindexing
 * move caching of schema to `solr.py`
 * implement on/off switch and configlet
+* error handling for stale connections
 * provide decorator for solr exceptions
 * what about http timeouts?
 * design and hook up searching
Index: collective.solr/trunk/collective/solr/indexer.py
===================================================================
--- collective.solr/trunk/collective/solr/indexer.py	(revision 13616)
+++ collective.solr/trunk/collective/solr/indexer.py	(revision 13617)
@@ -37,6 +37,7 @@
             data = self.getData(obj, attributes)
             self.prepareData(data)
             try:
+                logger.debug('indexing %r (%r)' % (obj, data))
                 conn.add(**data)
             except SolrException, e:
                 logger.exception('exception during index')
@@ -52,6 +53,7 @@
             # TODO: perhaps we should consider <uniqueKey> here
             assert data.has_key('id'), "no id in object data"
             try:
+                logger.debug('unindexing %r (%r)' % (obj, data))
                 conn.delete(id=data['id'])
             except SolrException, e:
                 logger.exception('exception during delete')
@@ -63,6 +65,7 @@
         conn = self.getConnection()
         if conn is not None:
             try:
+                logger.debug('committing')
                 conn.commit()
             except SolrException, e:
                 logger.exception('exception during commit')
@@ -103,6 +106,7 @@
 
     def closeConnection(self):
         """ close the current connection, if any """
+        logger.debug('closing connection')
         if getattr(self, '_v_connection', None) is not None:
             self._v_connection.close()
             self._v_connection = None
@@ -113,6 +117,7 @@
         conn = getattr(self, '_v_connection', None)
         if conn is None and self.host is not None:
             host = '%s:%d' % (self.host, self.port)
+            logger.debug('opening connection to %s' % host)
             conn = SolrConnection(host=host, solrBase=self.base, persistent=True)
             self._v_connection = conn
             self.schema = None
@@ -124,6 +129,7 @@
         if self.schema is None:
             conn = self.getConnection()
             if conn is not None:
+                logger.debug('getting schema from solr')
                 self.schema = conn.getSchema()
         return self.schema
 
